### PR TITLE
PP mock service

### PR DESCRIPTION
**Pull request by sk263**
**Description:**
This pull request introduces a basic functional partner plugin mock service to the CMB repository, intended to provide a simple implementation. The mock service simulates responses for defined partner service calls, offering static, meaningful example data.

**Changes:**
Added service handlers for:
- AccommodationProductList (v1, v2)
- AccommodationProductInfo (v1, v2)
- AccommodationSearch (v1, v2)
- Mint (v1, v2)
- Validation (v1, v2)

**Added caching:**
This caching strategy ensures that the price used at every stage of the booking process is consistent and validated, preventing discrepancies and potential issues.

**Search & Cache:** When a user call the AccomodationSearch service, the AccommodationSearchResult message is cached with search_id as the key.

**Validate & Cache:**  To validate a chosen accommodation, the search_id is sent to a validation service. This service:
Retrieves the cached search results using the search_id.
Extracts the PriceDetail from the results.
Caches the PriceDetail, this time associated with the validation_id as the key.

**Mint & Retrieve:** The validation_id is then passed to the "mint" service (likely the booking confirmation stage). This service retrieves the validated PriceDetail from the cache using the validation_id.

## Summary by CodeRabbit

- **New Features**
	- Comprehensive documentation for accommodation functionalities, including product lists, product info, and search criteria.
	- New JSON files providing structured examples for accommodation product info, search queries, and validation processes.
	- Introduction of minting services for both version 1 and version 2, enabling mint requests and validation services.
	- New ping service for health checks.

- **Bug Fixes**
	- Enhanced error handling and logging for various service methods.

- **Documentation**
	- Updated documentation files with detailed specifications and examples for accommodation-related functionalities.

- **Chores**
	- New Go packages and JSON structures added to support accommodation and minting services.

### CODE REVIEW

File: pp-mock/handlers/accommodation/v2/accommodation_search.go [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+						},
+					},
+				},
+			}, nil
+		}
+	}
+
+	searchResults := []*accommodationv2.AccommodationSearchResult{}
+	availableProperties := []*accommodationv2.PropertyExtendedInfo{}
+	var resultIDnum int32 = 1
```
[the remark this answers is not in the document]
Comment (sk263) [reply pp-mock/handlers/accommodation/v2/accommodation_search.go:72]: This is good, but not needed for mock service.

File: pp-mock/handlers/accommodation/v1/accommodation_search.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+							Month: query.TravelPeriod.GetEndDate().GetMonth(),
+							Day:   query.TravelPeriod.GetEndDate().GetDay(),
+						},
+					},
+					TravellerIds: getTravellerIDs(query.Travellers),
+					Beds:         room.Beds,
+					PriceDetail: &typesv1.PriceDetail{
+						Price: &typesv1.Price{
+							Value: "100",
+							Currency: &typesv1.Currency{
```
[the remark this answers is not in the document]
Comment (evlekht) [reply pp-mock/handlers/accommodation/v1/accommodation_search.go]: moved that to const
but having a function like that seems redundant, so function is not needed

File: pp-mock/handlers/accommodation/v2/accommodation_search.go [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+
+	// loop queries and check if there is travel period
+	for _, query := range req.Queries {
+		// TODO@ I would suggest to do 2 things to improve readability:
+		// TODO@ 1) add function that will cast TravelPeriod or Date to time.Time
+		// TODO@ 2) extract constants, move them as time.Time to some common package accessible by v1 and v2
+		// TODO@ 3) maybe even create a function in common package that will check if the date is in the allowed period
+		// only period between 01.06.2025 and 30.06.2025 is allowed - represents available period for the booking
+		if (query.TravelPeriod.GetStartDate().GetYear() != 2025 || query.TravelPeriod.GetStartDate().GetMonth() != 6 || query.TravelPeriod.GetStartDate().GetDay() <= 1) ||
+			(query.TravelPeriod.GetEndDate().GetYear() != 2025 || query.TravelPeriod.GetEndDate().GetMonth() != 6 || query.TravelPeriod.GetEndDate().GetDay() >= 30) {
```
[the remark this answers is not in the document]
Comment (evlekht) [reply pp-mock/handlers/accommodation/v2/accommodation_search.go]: thats cleaned up already

File: pp-mock/server.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+	reflection.Register(grpcServer)
+
+	port := 55555
+	var err error
+	p, found := os.LookupEnv("PORT")
+	if found {
+		port, err = strconv.Atoi(p)
+		if err != nil {
+			panic(err)
```
[the remark this answers is not in the document]
Comment (evlekht) [reply pp-mock/server.go:58]: lets ignore that for now

File: pp-mock/services/data/mock_data.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+var PropertiesV1 []*accommodationv1.PropertyExtendedInfo
+var PropertiesV2 []*accommodationv2.PropertyExtendedInfo
+
+func init() {
+	if err := json.Unmarshal(propertiesJSON, &PropertiesV1); err != nil {
+		panic(fmt.Errorf("error unmarshaling properties v1: %v", err))
+	}
+	if err := json.Unmarshal(propertiesJSON, &PropertiesV1); err != nil {
+		panic(fmt.Errorf("error unmarshaling properties v2: %v", err))
```
[the remark this answers is not in the document]
Comment (evlekht) [reply pp-mock/services/data/mock_data.go]: pushed fix, ty

File: pp-mock/handlers/accommodation/accommodation.go [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+)
+
+var (
+	allowedTimePeriodStart = time.Date(2025, time.June, 1, 0, 0, 0, 0, time.UTC)
+	allowedTimePeriodEnd   = time.Date(2025, time.June, 30, 0, 0, 0, 0, time.UTC)
+)
+
+func DateV1ToTime(date *typesv1.Date) time.Time {
+	return time.Date(int(date.GetYear()), time.Month(date.GetMonth()), int(date.GetDay()), 0, 0, 0, 0, time.UTC)
+}
```
[the remark this answers is not in the document]
Comment (evlekht) [reply pp-mock/handlers/accommodation/accommodation.go]: we'll never call this with nil date

File: pp-mock/handlers/accommodation/v1/accommodation_search.go [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+
+	response := &accommodationv1.AccommodationSearchResponse{
+		Header: &typesv1.ResponseHeader{
+			Status: typesv1.StatusType_STATUS_TYPE_SUCCESS,
+		},
+		Metadata: &typesv1.SearchResponseMetadata{
+			SearchId: &typesv1.UUID{Value: searchID},
+		},
+		Results: searchResults,
+	}
```
[the remark this answers is not in the document]
Comment (evlekht) [reply pp-mock/handlers/accommodation/v1/accommodation_search.go:181]: there is no need for caching, all will be in-mem

File: pp-mock/handlers/accommodation/accommodation.go [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+// only period between 01.06.2025 and 30.06.2025 is allowed - represents available period for the booking
+func IsTravelPeriodAllowed(travelPeriod *typesv1.TravelPeriod) bool {
+	startDate := DateV1ToTime(travelPeriod.GetStartDate())
+	endDate := DateV1ToTime(travelPeriod.GetEndDate())
+
+	return !startDate.Before(allowedTimePeriodStart) && !endDate.After(allowedTimePeriodEnd)
+}
```
[the remark this answers is not in the document]
Comment (evlekht) [reply pp-mock/handlers/accommodation/accommodation.go]: those are never nil

**Review by evlekht — APPROVED** [review]

**Pull request merged**